src/client.ts

```diff
@@ -1,34 +1,42 @@
-import { createPublicClient, createTestClient, createWalletClient, custom, http } from 'viem'
-import { mainnet, sepolia } from 'viem/chains'
-import { Chain } from './types/networks'
+import { createPublicClient, createWalletClient, custom, http, PublicClient, WalletClient } from 'viem'
+import { Chain, Chains, viemChains } from './types/networks'
+import { EthereumProvider, isEthereumProvider } from './types/provider'
 
-// @ts-ignore
-const mainnetClient = () => createWalletClient({ chain: mainnet, transport: custom(window.ethereum!) })
-// @ts-ignore
-const mainnetPublicClient = () => createPublicClient({ chain: mainnet, transport: custom(window.ethereum!) })
-
-// @ts-ignore
-const sepoliaClient = () => createWalletClient({ chain: sepolia, transport: custom(window.ethereum!) })
-// @ts-ignore
-const sepoliaPublicClient = () => createPublicClient({ chain: sepolia, transport: custom(window.ethereum!) })
+async function supportsPublicActions(provider: EthereumProvider): Promise<boolean> {
```

Comment:
this is a utility function to check if the provider allows public actions. just making a request to get the current block number